runtime/expr/cast.go

```diff
@@ -106,7 +107,8 @@ func (c *casterFloat16) Eval(ectx Context, val *zed.Value) *zed.Value {
 	if !ok {
 		return c.zctx.WrapError("cannot cast to float16", val)
 	}
-	return ectx.CopyValue(*zed.NewFloat16(float32(f)))
+	f16 := float16.Fromfloat32(float32(f))
+	return ectx.CopyValue(*zed.NewFloat16(f16.Float32()))
```

Comment:
Should there be a test for this?

Reply:
There is: TestCasts/time(float16(1589126400000000000) now fails without it.